### ISSUE TITLE
A deployer library

### ISSUE THREAD

**Issue by patricoferris** [ocurrent/ocurrent-deployer#141]
I've been building a (private) deployer and have ended up re-using a lot of components from `ocurrent-deployer`. I was wondering if there was any interest (or perhaps even some preliminary work :)) on extracting some of the reusable components into a publishable `current_deployer` library. In particular, things like https://github.com/ocurrent/ocurrent-deployer/blob/master/src/build.ml ?

**Comment by talex5** [comment]
Yes please!

**Comment by avsm** [comment]
Yes please!

**Comment by tmcgilchrist** [comment]
@patricoferris did you make any progress on this? Having re-usable components for deployer is a great idea. 
Mark has written AWS ones and I'm working on equivalent Azure ones for a personal project.

**Comment by patricoferris** [comment]
Unfortunately not far beyond re-implementing most of the `build.ml` file (I used records instead of first-class modules) and then providing some docker helper functions (including a docker-compose-based CI). But not much beyond what could be extracted here.

One really great feature would be private GitHub repository support, which I think such a deployer library would need. I'll add my voice to that issue if I can find (and if it exists, I think it does)

**Comment by tmcgilchrist** [comment]
Yes such an issue exists on both OCaml-CI and now on ocurrent. I had a
rough POC working using tokens last year but I’ve not had time to look at
it since.
